File: the_door/tests/fixtures/sample_codebases/typescript_simple/userService.ts
/**
 * User data access service.
 */
export class UserService {
    /** Find all users in the database */
    async findAll(): Promise<User[]> {
        return [];
    }

    /** Create a new user record */
    async create(data: CreateUserDto): Promise<User> {
        return { id: '1', ...data };
    }

    /** Delete a user by ID */
    async delete(id: string): Promise<void> {
        // TODO: Add soft-delete support
    }
}

File: the_door/tests/spike/fixtures/broken_integration/user_service.py
"""使用者服務：描述上「持久化使用者」，但實作只存在記憶體 —— 故意斷裂。
完全不 import / 不呼叫 Database。"""


class UserService:
    def __init__(self):
        self._users = []

    def save_user(self, user):
        # BUG（故意）：宣稱寫入資料庫，實際只 append 到記憶體 list
        self._users.append(user)
        return True
